### ISSUE TITLE
NPE (EMF-Deckard)

### ISSUE THREAD

**Issue by dmazinanian** [tsantalis/JDeodorant#41]
The symptoms are like #33: Calling PDGExpressionGap#getReturnType() for the only expression gap in the following clone pair returns Null. Consequently, CloneInfoCSVWriter is throwing NPE.

Group 137, Pair 1-2: org.eclipse.emf.ecore.util.EcoreValidator#parseTypeParameterList (row 256) and org.eclipse.emf.ecore.util.EcoreValidator#parseTypeArgumentList (row 257)

```
java.lang.NullPointerException
    at ca.concordia.jdeodorant.eclipse.commandline.cloneinfowriter.CloneInfoCSVWriter.writeCloneInfo(CloneInfoCSVWriter.java:156)
    at ca.concordia.jdeodorant.eclipse.commandline.Application.testRefactoring(Application.java:765)
    at ca.concordia.jdeodorant.eclipse.commandline.Application.start(Application.java:185)
    at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:673)
    at org.eclipse.equinox.launcher.Main.basicRun(Main.java:610)
    at org.eclipse.equinox.launcher.Main.run(Main.java:1519)
    at org.eclipse.equinox.launcher.Main.main(Main.java:1492)
```

**Comment by dmazinanian** [comment]
Please also check group 4523 of EMF-CCFinder after the fix, I think the same problem is happening there.

**Comment by dmazinanian** [comment]
Also group 5436 in EMF-CCFinder.

**Comment by tsantalis** [comment]
Fixed in commit [389c4745dde6d4cf3dc07fe0528d000ddb964443](https://github.com/tsantalis/JDeodorant/commit/389c4745dde6d4cf3dc07fe0528d000ddb964443)
